File: java/src/main/behavioral/state/ConcreteStateB.java
package behavioral.state;

// ConcreteStateB: Implements behavior specific to State B
public class ConcreteStateB implements State {
    @Override
    public void handleRequest(Context context) {
        System.out.println("ConcreteStateB: Handling request. Switching to ConcreteStateA.");
        context.setState(new ConcreteStateA());
    }
}

File: python/src/behavioral/state/concrete_state_b.py
from .state import State


class ConcreteStateB(State):
    """
    Concrete State B: defines different behavior and transitions.
    """

    def handle(self) -> str:
        return "State B: Handling request."

    def next_state(self) -> State:
        from .concrete_state_a import ConcreteStateA
        return ConcreteStateA()
